### PR TITLE
feat: add icons [4]

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Motivation

Icons that will be used over the project, but are being used immediately on Nano Contract UI.

>[!TIP]
>It is recommended to use the light theme to see the icons.

### Acceptance Criteria
- feat: add ActionDot icon
image

- feat: add CircleCheck icon
image

- feat: add CircleClock icon
image

- feat: add CircleError icon
image

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by andreabadesso — COMMENTED** [review]
What do you think about moving those to

`src/components/icons/**`?

**Review by andreabadesso — APPROVED** [review]

**Comment by andreabadesso** [comment]
After talking to @alexruzenhack on slack, I understood that icons are being moved to `src/components/icons/**` in another PR (https://github.com/HathorNetwork/hathor-wallet-mobile/pull/455)

**Comment by alexruzenhack** [comment]
> After talking to @alexruzenhack on slack, I understood that icons are being moved to `src/components/icons/**` in another PR (#455)

Confirmed. Thank you!

File: src/components/ActionDot.icon.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as React from 'react'
+import Svg, { G, Path, Defs, ClipPath } from 'react-native-svg'
+
+export const ActionDot = () => (
```
Comment (pedroferreira1) [inline src/components/ActionDot.icon.js:11]: Why don't you use `width`, `height`, `fill` as props like in the others?

Also, these default props values could be in a constants file inside the icons folder.

**Review by pedroferreira1 — COMMENTED** [review]
I think @andreabadesso's suggestion to move to a `components/icons` folder should be done in this PR.

File: src/components/ActionDot.icon.js (hunk 1 above)
Comment (alexruzenhack) [reply src/components/ActionDot.icon.js:11]: This icon doesn't have any variation on design and I believe it will not have in the future either.

Comment (alexruzenhack) [reply src/components/ActionDot.icon.js:11]: Regarding the default prop values, I can change.

Comment (pedroferreira1) [reply src/components/ActionDot.icon.js:11]: If the others are ready for any size/color change, I don't see any reasons for us not to prepare this one also.

Comment (alexruzenhack) [reply src/components/ActionDot.icon.js]: I did it: https://github.com/HathorNetwork/hathor-wallet-mobile/pull/453/commits/e51b92f9b1566251bb04436e8b364b17ce826f40

File: src/components/Icons/ActionDot.icon.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export const ActionDot = ({ size = DEFAULT_ICON_SIZE, color = COLORS.black }) => (
+  <Svg
+    xmlns='http://www.w3.org/2000/svg'
+    width={DEFAULT_ICON_SIZE}
+    height={DEFAULT_ICON_SIZE}
+    viewBox={getViewBox(size)}
+    transform={getScale(size, DEFAULT_ICON_SIZE)}
+    fill='none'
+  >
+    <G clipPath='url(#a)' opacity={0.7}>
```
Comment (pedroferreira1) [inline src/components/Icons/ActionDot.icon.js:23]: Does it make sense to have this opacity hardcoded? Or maybe receive this in the props?

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**